Engine/Settings.cs

```diff
@@ -540,24 +540,37 @@ private static object GetSafeValueFromExpressionAst(ExpressionAst exprAst)
                         return new object[0];
                     }
 
-                    PipelineAst pipelineAst = arrExprAst.SubExpression.Statements[0] as PipelineAst;
-                    if (pipelineAst == null)
+                    var listComponents = new List<object>();
+                    // Arrays can either be array expressions (1, 2, 3) or array literals with statements @(1 `n 2 `n 3)
+                    // Or they can be a combination of these
+                    // We go through each statement (line) in an array and read the whole subarray
+                    // This will also mean that @(1; 2) is parsed as an array of two elements, but there's not much point defending against this
+                    foreach (StatementAst statement in arrExprAst.SubExpression.Statements)
                     {
-                        throw CreateInvalidDataExceptionFromAst(arrExprAst);
-                    }
+                        var pipelineAst = statement as PipelineAst;
+                        if (pipelineAst == null)
```

Comment:
Using C#'s [is](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/is) statement would make the code shorter and easier to read here and below

Reply:
I actually much prefer `is` and remember trying to use it here and resorting to `as` when it wouldn't give me a scoped variable. Seems to work now, so I might just be *non compos mentis*.